tests/InferTypesFuzz.elm

```diff
@@ -0,0 +1,329 @@
+module InferTypesFuzz exposing (exprTyped)
+
+import AST.Canonical as Canonical
+import AST.Common.Literal exposing (Literal(..))
+import AST.Common.Located as Located
+import AST.Common.Type as Type exposing (Type)
+import Common.Types exposing (VarName(..))
+import Fuzz exposing (Fuzzer)
+import Random exposing (Generator)
+import Random.Extra as Random
+import Shrink exposing (Shrinker)
+import TestHelpers exposing (located)
+
+
+exprTyped : Type -> Fuzzer Canonical.LocatedExpr
+exprTyped targetType =
```

Comment:
I wonder about this name. Maybe `canonicalExpr` or `canonicalExprOfType`? If we don't plan to fuzz other expressions (Frontend / Typed), `exprOfType` would probably be the best. WDYT?

Reply:
also it's weird that this module is named InferTypesFuzz when what it exposes is only a `Canonical.Expr` fuzzer :thinking:

Reply:
I didn't think very hard about the module name. It's the fuzzing bits for `InferTypesFuzz`. Would the name make more sense if the fuzzed tests was moved inside this module? Or do we want to rename the module and keep it's scope?

Reply:
Yeah moving the fuzz tests here would work!